org.eclipse.xtext/src/org/eclipse/xtext/service/OperationCanceledManager.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2014, 2016 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.service;
+
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.xtext.util.CancelIndicator;
+
+/**
+ * A facade for managing and working with cancellation exceptions of different platforms.
+ *
+ * @author Sven Efftinge - Initial contribution and API
+ *
+ * @since 2.8
+ */
+public class OperationCanceledManager {
+	protected RuntimeException getPlatformOperationCanceledException(Throwable t) {
+		if (t instanceof OperationCanceledException) {
+			return (RuntimeException) t;
+		}
+		if (t instanceof OperationCanceledError) {
+			return ((OperationCanceledError) t).getWrapped();
+		}
+		return null;
+	}
+
+	public boolean isOperationCanceledException(Throwable t) {
+		return getPlatformOperationCanceledException(t) != null;
+	}
+
+	/**
+	 * Rethrows OperationCanceledErrors and wraps platform specific OperationCanceledExceptions. Does nothing for any
+	 * other type of Throwable.
+	 */
+	public void propagateAsErrorIfCancelException(Throwable t) {
+		if (t instanceof OperationCanceledError) {
+			throw (OperationCanceledError) t;
```

Comment:
Unnecessary cast

Reply:
It is necessary here. t is a Throwable, which could be a checked exception also.